Simulator/Generator/MapTile.py

```diff
@@ -76,11 +76,24 @@ def resolve_buildings(self):
                         max_x = x
                     if max_z < z:
                         max_z = z
+
                 for hole in building['geometry']['coordinates'][1:]:
-                    holes.append([self.translate_coord(hole_coord) for hole_coord in hole])
+                    holes.append(
+                        [lon_lat_to_grid(self.area.bottom_left,
+                                         self.area.resolution,
+                                         LongLatCoordinate(hole_coord[0], hole_coord[1]))
+                         for hole_coord in hole])
+
+                bounds = [Coordinate3D(min_x, 0, min_z),
+                          Coordinate3D(max_x, building['properties']['height'] / self.area.resolution, max_z)]
 
-                bounds = [Coordinate4D(min_x, 0, min_z, 0),
-                          Coordinate4D(max_x, building['properties']['height'], max_z, self.dimensions.t + 1000)]
+                top_right_grid = lon_lat_to_grid(self.area.bottom_left,
+                                                 self.area.resolution,
+                                                 self.area.top_right, )
```

Comment:
make an area property `dimension` or similar since this exact call is done multiple times